File: src/main/java/me/taubsie/dungeonhub/application/classes/HelpDisplay.java
package me.taubsie.dungeonhub.application.classes;

import me.taubsie.dungeonhub.application.enums.EmbedColor;
import me.taubsie.dungeonhub.kord.application.exceptions.CommandExecutionException;
import org.jetbrains.annotations.NotNull;

import java.util.HashMap;
import java.util.Map;

public record HelpDisplay(@NotNull String description, EmbedColor embedColor, @NotNull Map<String, String> fields) {
    public static HelpDisplay fromDescription(@NotNull String description) {
        return new HelpDisplay(description, EmbedColor.DEFAULT, new HashMap<>());
    }

    public static HelpDisplay fromException(CommandExecutionException commandExecutionException) {
        return new HelpDisplay(commandExecutionException.getMessage(), EmbedColor.NEGATIVE, new HashMap<>());
    }
}

File: src/main/java/me/taubsie/dungeonhub/kord/application/misc/HelpDisplay.kt
package me.taubsie.dungeonhub.kord.application.misc

import me.taubsie.dungeonhub.kord.application.exceptions.CommandExecutionException
import me.taubsie.dungeonhub.kord.application.enums.EmbedColor

@JvmRecord
data class HelpDisplay(val description: String, val embedColor: EmbedColor, val fields: Map<String, String>) {
    companion object {
        fun fromDescription(description: String): HelpDisplay {
            return HelpDisplay(description, EmbedColor.DEFAULT, HashMap())
        }

        fun fromException(commandExecutionException: CommandExecutionException): HelpDisplay {
            return HelpDisplay(commandExecutionException.message!!, EmbedColor.NEGATIVE, HashMap())
        }
    }
}